### PR TITLE
Cardano ADA (@ledgerhq/hw-app-ada)

### PR DESCRIPTION
**Pull request by DCRichards**
### Description

This adds support for Cardano ADA, compatible with the [Cardano ADA Ledger Application](https://github.com/HiddenField/ledger-cardano-app).

### Changes

* Added `packages/hw-app-ada` containing public API and its own set of self-contained integration tests, see README for details on running these.
* Updated README.md to reflect new package.
* Added snapshot tests for Cardano API and updated `snapshots.json` with expected payloads.

### Notes
Submitted on behalf of [HiddenField](https://www.hiddenfield.com/). Thank you to the team and all those involved in bringing this to the community.

### CODE REVIEW

File: packages/hw-app-ada/test/src/api/getWalletPublicKeyWithIndex.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
+      .then(res => check(res))
```
Comment (gre) [inline packages/hw-app-ada/test/src/api/getWalletPublicKeyWithIndex.js]: maybe you can use `async` function too & do

```
for (let i=0; i<20; i++) {
  check(await ada.getWalletPublicKeyWithIndex(index))
}
```

?

**Comment by gre** [comment]
i'll release a @ledgerhq/hw-app-ada 0.0.0 so the tests can pass

File: packages/hw-app-ada/test/src/api/getWalletPublicKeyWithIndex.js (hunk 1 above)
Comment (DCRichards) [reply packages/hw-app-ada/test/src/api/getWalletPublicKeyWithIndex.js]: Ah yes, good catch, they're left over from pre `async/await`. There's a few of those so I'll grab them all while I'm at it.

**Comment by DCRichards** [comment]
Thanks for the feedback @gre, appreciate it, I've updated to reflect that now. 

Re: the tests, I'm glad you like what we've done, we're very pleased with how robust they are and how useful they've been during development. Do get in touch if you want to discuss anything on them further.

**Pull request merged**